resolver/resolver.go

```diff
@@ -517,6 +519,29 @@ func (res *Resolver) RestEnumerate(req *restful.Request, resp *restful.Response)
 	}
 }
 
+// RestAXFR handles HTTP requests to turn the zone into a transferable format
+func (res *Resolver) RestAXFR(req *restful.Request, resp *restful.Response) {
+	records := res.records()
+
+	AXFRRecords := models.AXFRRecords{
+		SRVs: records.SRVs.ToAXFRResourceRecordSet(),
+		As:   records.As.ToAXFRResourceRecordSet(),
+	}
+	AXFR := models.AXFR{
+		Records:        AXFRRecords,
+		Serial:         atomic.LoadUint32(&res.config.SOASerial),
+		Mname:          res.config.SOAMname,
+		Rname:          res.config.SOARname,
+		TTL:            uint32(res.config.TTL),
+		RefreshSeconds: res.config.RefreshSeconds,
```

Comment:
something like this is really non-dns-standard. should we come up with a way to more cleanly separate dns from non-dns standard things?